### PR TITLE
server: fix global setting system.vm.public.ip.reservation.mode.strictness is not really dynamic

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

If the original value is `false`, and search build is configured without the condition. 
Now change the value to `true`, it will not get effective due to missing condition.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Review by rohityadavcloud — APPROVED** [review]
LGTM - not sure if this needs other explicit testing

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/network/IpAddressManagerImpl.java [hunk 1]
```diff
@@ -496,9 +496,8 @@ public boolean configure(String name, Map<String, Object> params) {
         AssignIpAddressSearch.and("dc", AssignIpAddressSearch.entity().getDataCenterId(), Op.EQ);
         AssignIpAddressSearch.and("allocated", AssignIpAddressSearch.entity().getAllocatedTime(), Op.NULL);
         AssignIpAddressSearch.and("vlanId", AssignIpAddressSearch.entity().getVlanId(), Op.IN);
-        if (SystemVmPublicIpReservationModeStrictness.value()) {
-            AssignIpAddressSearch.and("forSystemVms", AssignIpAddressSearch.entity().isForSystemVms(), Op.EQ);
-        }
+        AssignIpAddressSearch.and("forSystemVms", AssignIpAddressSearch.entity().isForSystemVms(), Op.EQ);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/IpAddressManagerImpl.java:499]: so now, whether the setting is true or false, we allow testing for it in the query!

**Review by DaanHoogland — APPROVED** [review]
lgtm

**Comment by DaanHoogland** [comment]
@weizhouapache does it make sense to merge this change with #7144 ?

**Comment by weizhouapache** [comment]
> @weizhouapache does it make sense to merge this change with #7144 ?

@DaanHoogland 
it is already in :facepalm: 
I totally forgot it ...

**Pull request merged**